dune

```diff
@@ -7,7 +7,16 @@
 )
 
 (env
-  (release (flags -safe-string -w +a-4-42-44-45-48-3-58-32-33)
+  (release (flags -safe-string -w +a-4-42-44-45-48-3-58-32-33-70)
            (ocamlc_flags)
-           (ocamlopt_flags))
+           (ocamlopt_flags)
+           (js_of_ocaml
+            (compilation_mode whole_program)
+            (flags --no-source-map --opt=2 --enable=use-js-string --target-env=browser)))
+
+  (dev     (flags (:standard -safe-string -w -32 -warn-error -a+8))
```

Comment:
Two questions:

1. adding this `dev` environment, should I change my usual programming inner loop?

   Namely, coding then building learn-ocaml with a bash alias that runs:
   `cd $dir/learn-ocaml && eval $(opam env) && opam install . --deps-only --locked && make && make opaminstall && learn-ocaml build --repo=demo-repository serve`

2. I had the impression that the dev profile had less warnings enabled than in the release profile ? but probably I'm mistaken

Reply:
Historically it was impossible to build in `dev` mode because (i) dune didn't allow to customise the jsoo flags manually, it was tied to the built-in profiles, and (ii) source-map was blowing on us and the compilation wouldn't terminate. Thankfully both of these are now fixed :)

In the new setting, the jsoo `--opt=2` flag makes release compilation quite a bit slower so I'd advise to use the dev mode when, well, developping 🤯 . The `Makefile` still defaults to `release` though, I'll add a `PROFILE ?= release` variable.
Then you could just run `make testrun PROFILE=dev`

2. hm it's quite possible, I haven't taken much attention to the specific warning flags ; I've copied them from another of my projects mainly to disable warn-error in dev mode which I find very hard to work with ("yes I know that this argument is unused, I just commented out the code, will you go on anyways ?" ;) )